src/abstractblocksparsearray/abstractblocksparsematrix.jl

```diff
@@ -1 +1,76 @@
 const AbstractBlockSparseMatrix{T} = AbstractBlockSparseArray{T,2}
+
+# SVD is implemented by trying to
+# 1. Attempt to find a block-diagonal implementation by permuting
+# 2. Fallback to AbstractBlockArray implementation via BlockedArray
+
+function eigencopy_oftype(A::AbstractBlockSparseMatrix, T)
```

Comment:
```suggestion
function eigencopy_oftype_blocksparse(A::AbstractMatrix, T)
```
to make it more general, i.e. handle wrapped `AbstractBlockSparseMatrix`. Looks like other functions could broaden their signatures as well (if we own them we could also just keep the name unchanged, but I like the naming convention `f_blocksparse` to show what type of object is expected).

Reply:
(Of course they could be `@interface ::AbstractBlockSparseArrayInterface` functions but we don't have to do that right now.)

Reply:
Would you be okay with me leaving the implementation as is, and immediately swapping it for `@interface`-based implementations in a follow-up PR?